### PR TITLE
Firts trial for functions

### PR DESCRIPTION
**Pull request by verobega**
@ClaviHaze

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ */
+
+    'use strict';
+var MarkDownIt = require('markdown-it'),
+    md = new MarkDownIt();
+let fs = require('fs');
+let path = require('path');
+let https = require('https');
+let http = require('http');
+const fetch = require('node-fetch');
```
Comment (ClaviHaze) [inline app.js:24]: Here you could use const not only on this one, but also on the ones before since their value is not gonna change

File: app.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            }
+        })
+        .catch(err => {
+            let result = `${uri} ${urls} is broken`
+            console.log(result);
+            statsLinks(result)
+        })
+}
+
+let newArr = []
```
Comment (ClaviHaze) [inline app.js:99]: This could be at the top with the rest of your global variables

File: app.js (hunk 1 above)
Comment (verobega) [reply app.js:24]: yeah, good point!

File: app.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    let uniqueLinks = links.filter(unique);
+    let howManyUnique = uniqueLinks.length
+    console.log(`Unique : ${howManyUnique}`);
+    function unique(value, index, self) {
+        return self.indexOf(value) === index;
+    }
+}
+
+lookingForFile()
+validation()
```
Comment (ClaviHaze) [inline app.js:123]: Just check if you can reduce the global variables if they're not necessary in more than one function. In general, it's a great code, keep it up!

File: app.js (hunk 2 above)
Comment (verobega) [reply app.js:99]: I still going to do a few changes and my intention is to became all the function in pure functions to pass the test.

File: app.js (hunk 3 above)
Comment (verobega) [reply app.js:123]: Thanks a lot! It's funny to share code with partners!

File: app.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+function getLinks(fileMd, uri) {
+    let expRegURL = /^https?:\/\/[\w\-]+(\.[\w\-]+)+[/#?]?.*$/
+    let newFile = fileMd.replace(/[\(\)]/g, " ");
+    let expRegText = /(?<=\[).+?(?=\])/
+    let space = " "
+    let arrNewFile = newFile.split(space)
+    let links = arrNewFile.filter(text => text.includes('http'))
+    statsBasic(links)
+   
+
```
Comment (ClaviHaze) [inline app.js:57]: It's not necessary to leave more than one blank space according to eslint. Also some semicolons are missing

**Pull request closed without merging**